### DISCUSSION TITLE
how to use pipx app in pyton script

### DISCUSSION THREAD

**Discussion by omega3** [pypa/pipx#1086] [Q&A] [answered]
For example instead of importing gtts installed system-wide as python module I have gtts-cli installed with pipx. How can I use it in python script?
Instead:
`from gtts import gTTS`
can I have something like:
`from pipx gtts-cli import gTTS`

Disclaimer: I don't have any knowledge of programming at all.

**Comment by dukecat0** [comment]
This isn't the correct usage. You should invoke it with command line like this: `gtts-cli 'hello' --output hello.mp3` if you installed `gtts` with pipx.

**Comment by omega3** [comment]
I know but I want the program installed with pipx to become part of a script so that the input is taken from csv, file processed with pipx installed program and be further processed with a script. I want to achieve what was done [here](https://github.com/pndurette/gTTS/discussions/432) (as an example) but with pipx installed programs, for example piper.
Can [virtual environment](https://docs.python.org/3/library/venv.html) be used here?

**Reply by dukecat0** [reply, marked as answer]
pipx is used to install command line applications, i.e. you can run them from the command line directly.

> Can [virtual environment](https://docs.python.org/3/library/venv.html) be used here?

I think in this case you should create a virtual environment, and install packages like piper and gtts into it.
